demonstrations/tutorial_local_cost_functions.py

```diff
@@ -0,0 +1,471 @@
+more robust to noise, and have more well-behaved gradients with no
+plateaus for shallow circuits.
+
+
+.. figure:: ../demonstrations/local_cost_functions/Cerezo_et_al_local_cost_functions.png
+   :align: center
+   :width: 50%
+   
+Reproduced from [#Cerezo2020].
+   
+In a quick explanation, a global cost function is the type that is traditionally used.
+Information from the entire measurement is used to analyze the result of the 
+circuit, and a cost function is calculated from this to quantify the circuits 
+performance. A local cost function only considers information from a few qubits, 
+and attempts to analyze the behavior of the entire circuit from this limited scope.
+
+
+Cerezo et al. also handily prove that these local cost functions are bounded by the 
+global ones, i.e. if a global cost function is formulated in the manner discribed
+by Cerezo, then its corrisponding local cost function will always be 
+less than or equal to the global.
+
+Imports
+--------------
+
+In this notebook, we investigate the effect of barren plateaus in
+variational quantum algorithms, and how they can be mitigated using
+local cost functions.
+"""
+
+
+import pennylane as qml
+from pennylane import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.ticker import LinearLocator, FormatStrFormatter
+
+
+######################################################################
+# Visualizing the problem
+# --------------
+#
+# To start, lets look at the task of learning the identity gate
+# across multiple qubits. This will help us visualize the problem and get
+# a sense of what is happening in the cost landscape.
+#
+# First we define a number of wires we want to train on. The work by
+# Cerezo et al. shows that circuits are trainible under certain regimes, so
+# how many qubits we train on will effect our results.
+
+wires = 6
+dev = qml.device("default.qubit", wires=wires, shots=10000, analytic=False)
+
+
+######################################################################
+# Next, we want to define our QNodes, containing our ansatz. For this
+# simple example, an ansatz that works well is simply a rotation along X,
+# and a rotation along Y, repeated across all the qubits.
+#
+# We will also define our cost functions here. Since we are trying to
+# learn the identity gate, a natural cost function is the probablity of measuring the 
+# zero state.
+#
+# .. math:: C = 1-p_{|0\rangle}
+#
+# We will apply this across all qubits for our global cost function, i.e.:
+#
+# .. math:: C_{G} = 1-p_{|00 \ldots 0\rangle}
+#
+# and only on one qubit for our local cost function:
+#
+# .. math:: C_{L} = 1-p_{|0>}.
+#
+# To implement this, we will define a separate QNode for the local cost
+# function and the global cost function.
+#
+
+
+def global_cost_simple(rotations):
+    for i in range(wires):
+        qml.RX(rotations[0][i], wires=i)
+        qml.RY(rotations[1][i], wires=i)
+    for i in range(wires-1):
+        qml.CNOT(wires=[i,i+1])    
```

Comment:
@Tstorwick it might be good to use the 'ring' pattern here; this ensures that the entanglement is non-local (and not classically simulatable by a tensor network simulator!)

E.g.:

```suggestion
    qml.broadcast(qml.CNOT, wires=wires, pattern="ring")
```

(see the broadcast documentation here: https://pennylane.readthedocs.io/en/stable/code/api/pennylane.broadcast.html)